# USING LINEAR REGRESSION TO PREDICT THE TIMING OF USAIN BOLT IN 200 M RACE ACCORDING TO AGE IN BETWEEN THE AGES HE WAS ACTIVELY PLAYING.

# THIS DATASET WAS CREATED BY ME ACCORDING TO THE DATA RETRIEVED FROM THE INTERNET.

In [94]:
import numpy as np
import pandas as pd

In [66]:
data=pd.read_csv("/content/usainbolt_200m.csv")

In [95]:
data.head(12)

AGE  YEAR  TIMING
0    16  2002   20.61
1    17  2003   20.43
2    18  2004   19.93
3    19  2005   20.03
4    21  2007   19.91
5    22  2008   19.30
6    23  2009   19.19
7    25  2011   19.40
8    26  2012   19.32
9    27  2013   19.66
10   29  2015   19.55
11   30  2016   19.78

In [96]:
x=data.iloc[:,:1]

x.head()

AGE
0   16
1   17
2   18
3   19
4   21

In [97]:
y=data.iloc[:,-1]

y.head()

0    20.61
1    20.43
2    19.93
3    20.03
4    19.91
Name: TIMING, dtype: float64

In [102]:
from sklearn.model_selection import train_test_split

x_train,x_test,y_train,y_test=train_test_split(x,y, test_size=0.35)

In [103]:
from sklearn.linear_model import LinearRegression

model=LinearRegression()

model.fit(x_train,y_train)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None, normalize=False)

In [104]:
prediction=model.predict(x_test)

prediction

array([19.51997712, 20.2195881 , 20.14185355, 19.2090389 , 19.83091533])

In [105]:
model.score(x,y)

0.4077014341519952

# THE ACCURACY OF THE MODEL IS 40.77%




In [106]:
data_new={'AGE':24}

index=[1]

In [107]:
usain_bolt_data=pd.DataFrame(data_new,index)

print(usain_bolt_data)

   AGE
1   24


In [108]:
prediction_new=model.predict(usain_bolt_data)

print("The timing prediction for the usain bolt at the given age is", prediction_new)

The timing prediction for the usain bolt at the given age is [19.67544622]


# HENCE THE PREDICTED TIMING FOR 200M RACE FOR USAIN BOLT AT THE AGE OF 24 IS 19.675 SECONDS.